lib/features/registration/pages/signup_page.dart

```diff
@@ -137,7 +137,14 @@ class _SignUpPageState extends State<SignUpPage> {
           builder: (context, state) {
             return isLoading
                 ? const Center(
-                    child: CircularProgressIndicator(),
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        Text('Hold on, we are creating your account...'),
+                        SizedBox(height: 16),
+                        CircularProgressIndicator(),
+                      ],
+                    ),
                   )
                 : _buildSignUpForm(locals, size, isUs);
```

Comment:
Also, these things I started putting them into separate widgets as it helps in terms in performance and also is not going to be re-rendered by the Engine